src/Samples/SarifTrim/Program.cs

```diff
@@ -0,0 +1,81 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Microsoft.CodeAnalysis.Sarif;
+using Microsoft.CodeAnalysis.Sarif.Writers;
+
+namespace SarifTrim
+{
+    /// <summary>
+    ///  SarifTrim is a one-off sample removing some redundant and some excessively large parts of a Sarif log
+    ///  for handling of a huge log.
+    /// </summary>
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            if (args.Length < 1)
+            {
+                Console.WriteLine(@"Usage: SarifTrim [inputFilePath] [outputFilePath?] [removeParts?]
+    removeParts: Comma-separated sections to remove. Options: UriBaseIds,CodeFlows,RelatedLocations,Graphs,GraphTraversals,Stacks,WebRequests,WebResponses");
```

Comment:
>RelatedLocations [](start = 83, length = 16)

RelatedLocations is a tricky thing to remove as messages have a mechanism for linking to them. i.e., a naive removal of related locations will result in messages with a link that can't be resolved. Might be safer to exclude this, alternately the code could raise an exception on determining that removal might invalidate some messages.

Reply:
Larry commented on this also. The core changes the consolidator makes should be meaning-preserving. When you ask it to remove sections wholesale, you are losing meaning in exchange for space savings. I want to make this possible in cases like Grant's where the log is just too big and you need a way to grapple with it.

I've added a class comment remark section on SarifConsolidator warning about this.